tests/unit/test_firewall.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Test cases for firewall module."""
+
+from ipaddress import IPv4Network
+
+import pytest
+
+from firewall import Firewall
+
+
+@pytest.mark.parametrize(
```

Comment:
Test cases:
* empty IP address range with no, one and multiple exclude
* single IP address range (so a range with just 1 IP address) with no, one and multiple exclude where for single and multiple one or more match including single IP and range for these
* single Ip address range with multiple IP addresses with no, single and multiple exclude where the exclude are single IP address or multiple IP address ranges
* similar for multiple IP address ranges

Reply:
Added, thank you!